### DISCUSSION TITLE
Section 6 Thunderloan - Lesson 39: Oracle Manipulation: Minimized - Some confusion in lesson video

### DISCUSSION THREAD

**Discussion by 0xharryriddle** [Cyfrin/security-and-auditing-full-course-s23#201] [Q&A] [answered]
I’m genuinely interested in this course. Thank you, Cyfrin, for creating it.

While learning lesson 39 in section 6 (Thunderloan), I found the order of actions in the flash loan activity confusing.

At 7:05 in the video, Mr. Collins mentioned that we could sell NFTs for plus 1 WETH  and then undo the swap. However, i think we have to first undo the swap to put the price back to normal (1WETH = 10USDC) before selling the NFTs to take profit (1WETH). 

![image](https://github.com/Cyfrin/security-and-auditing-full-course-s23/assets/100945007/3b55082b-6268-431c-adae-3fdb945d644e)

**Comment by EngrPips** [comment]
Hello @thai-cong-nguyen, If I remember correctly, we are manipulating the price of USDC on a decentralized exchange that `Whoops` NFT marketplace is reading `USDC` price from to buy the `NFT` at a lower price.  The point is the actual price of USDC across other true sources still remains valid even though we have manipulated the price on `T-Swap` exchange. So when we successfully buy the `NFT` on `Whoops` NFT marketplace, it doesn't matter if we undo the price we manipulated before selling our `NFT` or if we sell our `NFT` before undoing the price we manipulated as we will be selling our `NFT` to a platform that reads `USDC` price from a correct source. Please ask questions if this isn't clear enough.

**Reply by 0xharryriddle** [reply]
Hi sir, thanks for your reply. I'm not sure if I understood correctly. I believe that we buy the `NFT` with `10 USDC` after we manipulate the price, and then we will sell it for an additional `1 WETH` once the price returns to normal.

**Reply by EngrPips** [reply, marked as answer]
We don't need to wait for the price to return to normal on the `T-Swap` dexes because manipulating the price of `USDC` On `T-Swap` will not manipulate the price of `USDC` on Binance, CoinBase, or Uniswap pool.  We bought `NFT` for `10USDC`, which is supposed to be equivalent to `1 WETH`, but because we manipulated the price on `T-Swap` dex, we paid weth amount that is very small, and we can go ahead and sell the NFT on any other platform where it will be worth `1 WETH`, so basically we used around 0.00001 WETH to buy the NFT  on `Whoops` marketplace, and we went ahead to sell it for 1 WETH on some other platform like Opensea. Please don't hesitate to ask questions if anything is still confusing.

**Reply by 0xharryriddle** [reply]
Oh, I was definitely clear. Thank you, sir.

**Reply by EngrPips** [reply]
Awesome, I am glad you figured it out. Keep crushing it, Friend.
